nixos/modules/services/games/vrising-server.nix

```diff
@@ -0,0 +1,174 @@
+{ config, lib, pkgs, ... }:
+with lib;
+
+let
+  cfg = config.services.vrising-server;
+  varLibStateDir = "/var/lib/${cfg.stateDir}";
+in {
+  options = {
+    services.vrising-server = {
+      enable = mkEnableOption "V Rising dedicated server";
+
+      stateDir = mkOption {
+        type = types.str;
+        default = "vrising-server";
+        description = ''
+          Directory to store all stateful configuration, logs and save files.
+          Will be created as subdirectory of <literal>/var/lib/</literal>.
+        '';
+      };
+
+      passwordFile = mkOption {
+        type = types.path;
+        description = ''
+          File that contains the password for the server.
+        '';
+      };
+
+      openFirewall = mkEnableOption ''
+        Whether to open the ports for the server defined in hostSettings in the firewall.
+      '';
+
+      hostSettings = mkOption {
+        type = types.submodule {
+          freeformType = (pkgs.formats.json { }).type;
+
+          options = {
+            port = mkOption {
+              type = types.port;
+              default = 9876;
+              description = ''
+                Port for the server, will be added to allowedUDPPorts.
+              '';
+            };
+
+            queryPort = mkOption {
+              type = types.port;
+              default = 9877;
+              description = ''
+                Query port for the server, will be added to allowedUDPPorts.
+              '';
+            };
+          };
+        };
+        default = { };
+        example = literalExpression ''
+          {
+            Name = "V Rising Server";
+            Description = "";
+            MaxConnectedUsers = 5;
+            MaxConnectedAdmins = 5;
+            ServerFps = 30;
+            SaveName = "world1";
+            Secure = true;
+            ListOnMasterServer = false;
+            AutoSaveCount = 20;
+            AutoSaveInterval = 300;
+            GameSettingsPreset = "StandardPvE";
+            AdminOnlyDebugEvents = true;
+            DisableDebugEvents = false;
+          }'';
+        description = ''
+          Host settings as defined in <link xlink:href="https://github.com/StunlockStudios/vrising-dedicated-server-instructions">this repository</link>.
+          Look at the reference to see what you can configure in this attribute set.
+
+          Do not set <literal>Password</literal> here but use the dedicated <literal>passwordFile</literal> option instead.
+        '';
+      };
+
+      gameSettings = mkOption {
+        type = with types; attrsOf anything;
```

Comment:
this one should also use the json type, but a freeform module proably won't be necessary.